spec/support/database.rb

```diff
@@ -22,19 +32,45 @@ def setup
     # log.sev_threshold = Logger::DEBUG
     log.level = Logger::Severity::UNKNOWN
     ActiveRecord::Base.logger = log
+
+    @connection_options = YAML.safe_load(ERB.new(IO.read("#{dirname}/database.yml")).result)[self.class.adapter]
+
     self
   end
 
   def migrate
     ActiveRecord::Migration.verbose = false
-    databaseyml = YAML.safe_load(ERB.new(IO.read("#{dirname}/database.yml")).result)
-    ActiveRecord::Base.establish_connection databaseyml[adapter]
+
+    ensure_database
 
     require "#{dirname}/schema"
     require "#{dirname}/models"
 
     self
   end
-end
 
-Database.new.setup.migrate
+  private
+
+  def ensure_database
+    database_exists?
+  rescue ActiveRecord::NoDatabaseError
+    create_database
+    database_exists?
+  end
+
+  def database_exists?
+    ActiveRecord::Base.establish_connection(@connection_options)
+    !!ActiveRecord::Base.connection
+  end
+
+  def create_database
+    database = @connection_options.delete("database")
+    ActiveRecord::Base.establish_connection(@connection_options)
```

Comment:
I tend to create the database outside of here. In my mind, the database admin is a different concern than the application itself / the test suite.

Locally I have a `bin/setup` which does a createdb.
It wasn't nailed down so I didn't get it up to snuff.

which creates the DB in much the same way that `.github/workflows/ci.yml` does (or at least used to)
```

psql -c 'create database virtual_attributes;' || echo 'db exists'
mysql -u root -e 'CREATE SCHEMA IF NOT EXISTS 'virtual_attributes';
```

Reply:
Damn I totally didn't see the bin/setup.  It's not being called in CI (I checked ci.yaml for how to set things up).  Ok, if that exists, I might just drop the database setup bit entirely and just doc that in the README.

Reply:
Ok I removed this DB setup entirely, preferring bin/setup made a cleanup PR for it: #126